tools/buildsteps/windows/BuildSetup.bat

```diff
@@ -1,319 +1,319 @@
-    copy /Y %%~dpna.cer %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.cer
-    copy /Y %%~dpna.appxsym %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.appxsym
-    goto END
-  )
-  rem Release builds don't have Release in it's name
-  for /F %%a IN ('dir /B /S %WORKSPACE%\AppPackages ^| findstr /I /R "%APP_NAME%_.*_%TARGET_ARCHITECTURE%\.appx$"') DO (
-    copy /Y %%a %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.appx
-    copy /Y %%~dpna.cer %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.cer
-    copy /Y %%~dpna.appxsym %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.appxsym
-    goto END
-  )
-
-  rem apxx file has win32 instead of x86 in it's name
-  if %TARGET_ARCHITECTURE%==x86 (
-    for /F %%a IN ('dir /B /S %WORKSPACE%\AppPackages ^| findstr /I /R "%APP_NAME%_.*_win32_%buildconfig%\.appx$"') DO (
-      copy /Y %%a %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.appx
-      copy /Y %%~dpna.cer %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.cer
-      copy /Y %%~dpna.appxsym %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.appxsym
-      goto END
-    )
-
-    rem Release builds don't have Release in it's name
-    for /F %%a IN ('dir /B /S %WORKSPACE%\AppPackages ^| findstr /I /R "%APP_NAME%_.*_win32\.appx$"') DO (
-      copy /Y %%a %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.appx
-      copy /Y %%~dpna.cer %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.cer
-      copy /Y %%~dpna.appxsym %base_dir%\%APP_NAME%-%GIT_REV%-%BRANCH%-%TARGET_ARCHITECTURE%.appxsym
-      goto END
-    )
-  )
-
-:DIE
-  ECHO ------------------------------------------------------------
-  ECHO !-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-
-  ECHO    ERROR ERROR ERROR ERROR ERROR ERROR ERROR ERROR ERROR
-  ECHO !-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-!-
-  set DIETEXT=ERROR: %DIETEXT%
-  echo %DIETEXT%
-  SET exitcode=1
-  ECHO ------------------------------------------------------------
-  GOTO END
-
-:END
-  IF %promptlevel% NEQ noprompt (
-    ECHO Press any key to exit...
-    pause > NUL
-  )
-  EXIT /B %exitcode%
+@ECHO OFF
+REM SETLOCAL ENABLEDELAYEDEXPANSION
+REM setup all paths
+PUSHD %~dp0\..\..\..
+SET base_dir=%CD%
+POPD
+SET builddeps_dir=%base_dir%\project\BuildDependencies
+SET bin_dir=%builddeps_dir%\bin
+SET msys_dir=%builddeps_dir%\msys64
+IF NOT EXIST %msys_dir% (SET msys_dir=%builddeps_dir%\msys32)
+SET sed_exe=%msys_dir%\usr\bin\sed.exe
+
+REM read the version values from version.txt
+SET version_props=^
+APP_NAME ^
+COMPANY_NAME ^
+PACKAGE_DESCRIPTION ^
+PACKAGE_IDENTITY ^
+PACKAGE_PUBLISHER ^
+VERSION_MAJOR ^
+VERSION_MINOR ^
+VERSION_TAG ^
+VERSION_CODE ^
+WEBSITE
+
+FOR %%p IN (%version_props%) DO (
+  FOR /f "delims=" %%v IN ('%sed_exe% -n "/%%p/ s/%%p *//p" %base_dir%\version.txt') DO SET %%p=%%v

```

Comment:
`^M` shouldn't be here

Reply:
yeah I think it got a bit messed up because I had turned off autocrlf in git

Reply:
hmm I've squashed it and looked at the end of line characters for everything and this line has the same crlf as every other line but for some reason github shows the ^M character.